### PR TITLE
feat: beta updater

### LINKED ISSUE #1169: ci: Updater for beta release
We have already supported updater for stable and nightly. It would be nice if we can support beta release also.

### PR DESCRIPTION
**Pull request by vansangpfiev**
## Describe Your Changes
Support beta updater, for now we have:
- Stable: retrieves only the latest version.
- Beta: the latest version is default, specific versions can be retrieved using the -v flag.
- Nightly: the latest version is default, specific versions can be retrieved using the -v flag.

## Fixes Issues

- Closes https://github.com/janhq/cortex.cpp/issues/1169

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/main.cc [hunk 1]
```diff
@@ -131,6 +133,13 @@ void ForkProcess() {
 int main(int argc, char* argv[]) {
   { file_manager_utils::CreateConfigFileIfNotExist(); }
 
+  // Delete temporary file if it exists
+  auto temp =
+      file_manager_utils::GetExecutableFolderContainerPath() / "cortex_temp";
+  if (std::filesystem::exists(temp)) {
+    std::filesystem::remove(temp);
```
Comment (namchuai) [inline engine/main.cc]: nit, maybe add a try catch here.

File: engine/commands/cortex_upd_cmd.cc [hunk 2]
```diff
@@ -61,69 +64,83 @@ bool CortexUpdCmd::GetStableAndBeta(const std::string& v) {
...
+    CTL_ERR("Unsupported OS or architecture: " << system_info.os << ", "
+                                               << system_info.arch);
+    return false;
+  }
+  CTL_INF("OS: " << system_info.os << ", Arch: " << system_info.arch);
+
+  // Download file
+  constexpr auto github_host = "https://api.github.com";
+  std::ostringstream release_path;
+  release_path << "/repos/janhq/cortex.cpp/releases";
```
Comment (dan-menlo) [inline engine/commands/cortex_upd_cmd.cc]: @vansangpfiev Just a note, we may need to support multiple download sources in the future (e.g. Github, and mirrors in Gitlab, Gitee)

**Review by dan-menlo — APPROVED** [review]
@vansangpfiev Approving to unblock. 

Can I check my high-level understanding:

- We are breaking off beta updater from stable (previous)
- We are currently hard-coding Github URLs 

Just FYI, we likely will support multiple download mirrors in the future, e.g. Github, Gitlab, Gitee

File: engine/commands/cortex_upd_cmd.cc [hunk 3]
```diff
@@ -23,93 +23,109 @@ void CortexUpdCmd::Exec(std::string v) {
...
-    if (!GetStableAndBeta(v))
+    if (!GetNightly(v))
       return;
   }
   CLI_LOG("Update cortex sucessfully");
 }
 
-bool CortexUpdCmd::GetStableAndBeta(const std::string& v) {
+bool CortexUpdCmd::GetStable(const std::string& v) {
+  // Check if the architecture and OS are supported
```
Comment (namchuai) [inline engine/commands/cortex_upd_cmd.cc]: I think we can remove this comment. Since we already have a check at `main.cc`

Comment (namchuai) [reply engine/commands/cortex_upd_cmd.cc]: Just nits. Not required for this commit to be merged.

File: engine/commands/cortex_upd_cmd.cc [hunk 4]
```diff
@@ -153,8 +226,7 @@ bool CortexUpdCmd::GetNightly(const std::string& v) {
   CTL_INF("Engine release path: " << url_parser::FromUrl(url_obj));
 
   std::filesystem::path localPath =
-      file_manager_utils::GetExecutableFolderContainerPath() / "cortex" /
-      path_list.back();
+      std::filesystem::temp_directory_path() / "cortex" / path_list.back();
```
Comment (namchuai) [inline engine/commands/cortex_upd_cmd.cc:229]: I tested, and we will need to create the corresponding directory, or we will have a crash.

```
  try {
    std::filesystem::create_directories(localPath.parent_path());
  } catch (const std::filesystem::filesystem_error& e) {
    CTL_ERR("Failed to create directories: " << e.what());
    throw;
  }
```

Comment (vansangpfiev) [reply engine/commands/cortex_upd_cmd.cc:229]: Thanks. Added `create_directories` if it does not exist.

**Review by namchuai — APPROVED** [review]
✅ Great work! Thank you!

**Pull request merged**